### ISSUE TITLE
[User Story] 3rd party password manager apps support

### ISSUE THREAD

**Issue by abodea** [mozilla-mobile/fenix#9773]
## Notes
This is a follow up with the current situation regarding more password manager apps.
Initial [bug](https://github.com/mozilla-mobile/fenix/issues/87) closed now.

## User Story
- As a user, I want to be able to use the password manager apps.

## Dependencies
Most popular password manager apps:
- [ ] [My passwords - Password Manager](https://play.google.com/store/apps/details?id=com.er.mo.apps.mypasswords&hl=en)
- [ ] [LastPass Password Manager](https://play.google.com/store/apps/details?id=com.lastpass.lpandroid&hl=en)
- [x] [Password Safe - Secure Password Manager](https://play.google.com/store/apps/details?id=com.reneph.passwordsafe&hl=en)
- [ ] [Password Manager SafeInCloud](https://play.google.com/store/apps/details?id=com.agilebits.onepassword&hl=en)
- [x] [1Password - Password Manager and Secure Wallet](https://play.google.com/store/apps/details?id=com.safeincloud.free&hl=en)
- [x] [Avast Passwords](https://play.google.com/store/apps/details?id=com.avast.android.passwordmanager&hl=en)
- [x] [Bitwarden Password Manager](https://play.google.com/store/apps/details?id=com.x8bit.bitwarden&hl=en)
- [x] [Enpass Password Manager](https://play.google.com/store/apps/details?id=io.enpass.app&hl=en_US)
- [x] [Lockwise](https://play.google.com/store/apps/details?id=mozilla.lockbox&hl=en)
- [x] [Password Keeper (Black Berry](https://play.google.com/store/apps/details?id=com.callpod.android_apps.keeper&hl=en)

### Acceptance Criteria
- [ ] Password manager app can access Fenix
- [ ] Fenix fills out logins automatically
- [ ] Fenix fills out logins automatically after the password manager app was changed
- [ ] Focus app parity

Below I will update the table once I will move forward with the testing and I will add information regarding what issues need to be fixed/investigated if it will be the case to contact the password manager apps developers.

| App | Autofill? | Bugs  | Comments  | 
| :--: | :--: | :---------: | :--: |
| My passwords | |       |          | 
| LastPass | |   https://github.com/mozilla-mobile/fenix/issues/4205     |        | 
| Password Safe  | N/A |  -     |      Note that this is offline storage for accounts, I was able to copy/paste any account in Fenix, works as expected      | 
| SafeInCloud |❌ |   https://github.com/mozilla-mobile/fenix/issues/10621    |         | 
| 1Password|   ✅  |   - |    autofill can be done without problems|    
| Avast Passwords | ❌ |    https://github.com/mozilla-mobile/fenix/issues/10628, https://github.com/mozilla-mobile/fenix/issues/10922   |          |
| Bitwarden  |     ✅  |  -     |  autofill can be done without problems  
| Lockwise |    ✅    |   https://github.com/mozilla-mobile/fenix/issues/9905, https://github.com/mozilla-mobile/fenix/issues/3511 | autofill can be done without problems     |
| Enpass  |   ✅     | https://github.com/mozilla-mobile/fenix/issues/10640      | autofill can be done without problems
| Password Keeper (Black Berry)|   ✅     | -     | autofill can be done without problems

| General bugs | 
| :-----------------: | 
| https://github.com/mozilla-mobile/fenix/issues/9854 |

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-16714)

**Comment by SS1113** [comment]
I use [Enpass Password Manager](https://play.google.com/store/apps/details?id=io.enpass.app&hl=en_US), is there going to be support for that as well?

**Comment by cadeyrn** [comment]
Please also test with Firefox Lockwise. It's not the most popular password manager, but important for other obvious reasons. :)

**Comment by abodea** [comment]
> I use [Enpass Password Manager](https://play.google.com/store/apps/details?id=io.enpass.app&hl=en_US), is there going to be support for that as well?
Hi @SS1113 I will include that one as well, thank you for the recommendation!

**Comment by abodea** [comment]
> Please also test with Firefox Lockwise. It's not the most popular password manager, but important for other obvious reasons. :)

Hi @cadeyrn I was going to include Lockwise too, I'm working on Lockwise too so I test that more often.

**Comment by hakkikaancaliskan** [comment]
@abodea #7742

**Comment by SS1113** [comment]
Enpass password manager still doesn't work on all my 4 devices, don't know why it is still marked as working in this issue.

Relevant issue: https://github.com/mozilla-mobile/fenix/issues/10640

**Comment by abodea** [comment]
I will re-check it @SS1113 and come back with some info, I had no issues with Enpass. 
Can you please tell me what build are you using? From Play Store I suppose?

**Comment by SS1113** [comment]
> I will re-check it @SS1113 and come back with some info, I had no issues with Enpass. 
> Can you please tell me what build are you using? From Play Store I suppose?


**`Nightly 200706 06:01`** it's from the Taskcluster. The build that used to be Firefox Preview but is now Firefox Nightly.

**Comment by ShatteredPixel** [comment]
> 
> 
> > I will re-check it @SS1113 and come back with some info, I had no issues with Enpass.
> > Can you please tell me what build are you using? From Play Store I suppose?
> 
> **`Nightly 200706 06:01`** it's from the Taskcluster. The build that used to be Firefox Preview but is now Firefox Nightly.

abodea, The above deleted account is mine. Please let me know if I can provide you any more information.

**Comment by ShatteredPixel** [comment]
>**`Nightly 200706 06:01`** it's from the Taskcluster. The build that used to be Firefox Preview but is now Firefox Nightly.

abodea, I am the reporter of this issue; Enpass Password Manager not working. Please let me know if I can provide you any more information. Thanks. :)

**Comment by ekager** [comment]
@abodea could we possibly test LastPass (and one other password manager that worked from comment 0 to compare) on the migrated release, beta, and nightly channels?

**Comment by abodea** [comment]
> @abodea could we possibly test LastPass (and one other password manager that worked from comment 0 to compare) on the migrated release, beta, and nightly channels?

Hi, @ekager tomorrow I will start testing those and add the results here!

**Comment by hakkikaancaliskan** [comment]
> > @abodea could we possibly test LastPass (and one other password manager that worked from comment 0 to compare) on the migrated release, beta, and nightly channels?
> 
> Hi, @ekager tomorrow I will start testing those and add the results here! 

Please don't forget to test #7742 too :)

**Comment by abodea** [comment]
I re-tested this for more details please check the following:

| App | Browser | Autofill?| Bugs| Comments  |  GV | 
| :--: | :--: | :---------: | :--: | :--: | :--: | 
| 1Password |Migrated RC (**latest**) | ✅    |     -    |      -       |      **79.0.2**      | 
| 1Password |Latest **Beta** migration | ✅    |     -    |      -       |      **80.0.0-beta2**      | 
| 1Password |Latest **Debug** (from master) | ✅    |     -    |      -       |      **81.0a1**      | 
| 1Password |**Brave** | ✅    |     -    |      -       |      N/A     | 
| 1Password |**Chrome**| ✅    |     -    |      -       |      N/A     | 
| LastPass |Migrated RC (**latest**) | ❌    |     -    |      -       |      **79.0.2**      | 
| LastPass |Latest **Beta** migration | ❌    |     -    |      -       |      **80.0.0-beta2**      | 
| LastPass |Latest **Debug** (from master) | ❌    |     -    |      -       |      **81.0a1**      | 
| LastPass |**Brave** | ✅    |     -    |      -       |      N/A     | 
| LastPass |**Chrome**| ✅    |     -    |      -       |      N/A     | 

Note that for LastPass sometimes the autofill pop up is not even displayed.
And when I was able to trigger the autofill pop up from LastPass, the same issue described here https://github.com/mozilla-mobile/fenix/issues/4205 is present in all 3 builds.

Example from **latest debug build from master.**
![Screenshot_20200804-143144](https://user-images.githubusercontent.com/42831109/89289747-6f41a900-d660-11ea-9305-c9a57e7afbf8.png)


Example from **Brave.**
![Screenshot_20200804-143152](https://user-images.githubusercontent.com/42831109/89289730-6a7cf500-d660-11ea-8220-376a18422136.png)

Lastpass Gif
![20200804-142439](https://user-images.githubusercontent.com/42831109/89290367-70270a80-d661-11ea-992e-4429a0ade6dd.gif)

**Comment by ekager** [comment]
Thanks so much @abodea !

**Comment by FrostedIce339** [comment]
Hi, @abodea I'm facing problems while using autofill using Enpass on my device but the issue description says that there's no problem with it. Is there any workaround that I can use to make it work? Or am I missing something? 

The pop-up dialogue seem to be continuously appearing and disappearing just as in the issue description that you linked #10640.

**Comment by abodea** [comment]
> Hi, @abodea I'm facing problems while using autofill using Enpass on my device but the issue description says that there's no problem with it. Is there any workaround that I can use to make it work? Or am I missing something?
> 
> The pop-up dialogue seem to be continuously appearing and disappearing just as in the issue description that you linked #10640.

Hi, @FrostedIce339 thank you for your comment.
If you check https://github.com/mozilla-mobile/fenix/issues/10640#issuecomment-629135567 my comment I also attached a GIF while I auto-filled with Enpass. Based on that I added the status on my bug description but I still added the bug submitted so we can keep the track of it. 
I will investigate and try to auto-fill again with Enpass and get back to you. 
Until then can you please let me know what device and on what build are you trying to reproduce this issue? (please attach a screenshot with about).

**Comment by abodea** [comment]
### Hi, @vesta0 I re-tested the last version of LastPass for more details please check the following:

| App | Browser | Autofill?| Bugs| Comments  |  GV | 
| :--: | :--: | :---------: | :--: | :--: | :--: | 
| **LastPass** |Migrated **RC** 79.0.4 (**latest**) | ❌    |    https://github.com/mozilla-mobile/fenix/issues/12128   |     As described [here ](https://github.com/mozilla-mobile/fenix/issues/12128#issue-648181999) the LastPass doesn't do anything, there is an error message: "Please scroll the browser window until the website address is visible, then try again." error is displayed    |      **79**      | 
| **LastPass** |Latest **Beta** migration 81.0.0-beta.1 | ❌    |    https://github.com/mozilla-mobile/fenix/issues/12128   |     As described [here ](https://github.com/mozilla-mobile/fenix/issues/12128#issue-648181999) the LastPass doesn't do anything, there is an error message: "Please scroll the browser window until the website address is visible, then try again." error is displayed   |       **80**     |   
| **LastPass** |Latest **Nightly**  | ✅    |     -    |      The autofill pop-up is displayed only when tapping on the LastPass app from the notification bar (for more details please check the attachments below)       |      **81.0a1**      | 

**NOTE**: 
- Right now the only way to make the LastPass pop-up appear in Fenix is to tap on the LastPass app from the notification bar.
And I was able to do this only when I enabled the Legacy autofill from the LastPass app (this will add the LastPass on the notification bar and after tapping on it the autofill pop-up will be displayed in Fenix Nightly).
![Screenshot_20200813-132839](https://user-images.githubusercontent.com/42831109/90124734-8d4a9000-dd69-11ea-9010-371b6ae135df.png)

- I tested with **Chrome**, **Brave**, and many other browsers and when I tapped on the username/password field the LastPass appeared without problems.
Below you can see that right after I tapped on the `username` field in Brave browser, the LastPass pop up was displayed.
![image](https://user-images.githubusercontent.com/42831109/90122393-e31d3900-dd65-11ea-98e9-b8b812ce7f83.png)

Regarding https://github.com/mozilla-mobile/fenix/issues/4205 I couldn't reproduce the issue, I will keep investigating it, but I do have Mozilla accounts saved on my LastPass account and I was able to autofill in the latest Nightly without problems on Twitter/Fb.

**Comment by jamy015** [comment]
> And I was able to do this only when I enabled the Legacy autofill from the LastPass app (this will add the LastPass on the notification bar and after tapping on it the autofill pop-up will be displayed in Fenix Nightly).

Even legacy autofill isn't working for me. The autofill pop-up just doesn't show at all after tapping the LastPass icon in the quick settings drawer.

79.0.5 (Build #2015758619)
AC: 48.0.15, 8755ac0cd
GV: 79.0.1-20200813192915
AS: 61.0.10

**Comment by vesta0** [comment]
We are working with LastPass to fix this issue. LastPass has not yet implemented support for the new autofill API but they are working on it. @abodea we should test this again in a month.

**Comment by tstackhouse** [comment]
LastPass has had support for the Android Oreo auto fill API for quite a long time.  My impression was that this was something related to Fenix not properly interacting with that system level API to allow any compatible app to provide autofills...

**Comment by serovar** [comment]
1Password does not recognize anymore the websites you are in on Nightly.

**Comment by kbrosnan** [comment]
It may be as simple as Lastpass is forcing Firefox into their legacy implementation and need to add `org.mozilla.firefox`, `org.mozilla.firefox_beta` and `org.mozilla.fenix` to their modern autofill whitelist. We have talked with them and they are looking into it from their side. Considering there are several other password mangers that work with Firefox and we have not heard any issues from those companies/user it seems reasonable that the problem is on Lastpass to investigate.

**Comment by liuche** [comment]
Also seeing a user report from #14633 for Keepass, which isn't in this list.

**Comment by sarumont** [comment]
@kbrosnan Is there any open issue with Lastpass that we could follow / vote for / bug them about? :) Still not working for me, so it appears that they are a bit slow...

**Comment by tgsachse** [comment]
Also would like to follow this thread ^

**Comment by Djfe** [comment]
I planned on posting this on #10640
but since this seems to affect lots of autofill apps and lots of devices (see #10640 and #14633), I'll comment on here instead:

> The issue is present in Firefox Beta 86 (beta2) while it was fixed in Nightly 86.
Same again:
Flickering fixed in Nightly 88, bug still happening beta 1 87.0.0

While flickering is fixed in Nightly, positioning for form inputs is still off (viewport is adjusted left/top position plus zoom level when you click on an `<input>`, but enpass and other form fill apps get transmitted the wrong position for them, so they render on top of your keyboard app etc.)
I'll upload videos now to showcase this (beta and nightly so you have the comparison).

For some reason flickering never affects nightly, maybe it's due to build settings? It's the most annoying part of this bug.
Sure you could use your own proprietary API (better since it works automatically), but it would be nice if this would work on system level like it does for all other apps 
and browsers.

I'm on Android 10 Havoc OS
OnePlus 3T

**Comment by Djfe** [comment]
Beta:
https://youtu.be/6JW8j1NG9ms

Nightly:

**Comment by Djfe** [comment]
Beta:
https://youtu.be/6JW8j1NG9ms
[![Beta Screenscreenrecording](http://img.youtube.com/vi/YOUTUBE_VIDEO_ID_HERE/0.jpg)](http://www.youtube.com/watch?v=YOUTUBE_VIDEO_ID_HERE "Video Title")

Nightly:
[![Nightly ](http://img.youtube.com/vi/YOUTUBE_VIDEO_ID_HERE/0.jpg)](http://www.youtube.com/watch?v=YOUTUBE_VIDEO_ID_HERE "Video Title")
https://youtu.be/pLSv8YjuMj8

**Comment by Djfe** [comment]
Beta:
[![Beta Screenrecording](http://img.youtube.com/vi/6JW8j1NG9ms/0.jpg)](http://www.youtube.com/watch?v=6JW8j1NG9ms "Beta")

Nightly:
[![Nightly Screenrecording](http://img.youtube.com/vi/pLSv8YjuMj8/0.jpg)](http://www.youtube.com/watch?v=pLSv8YjuMj8 "Nightly")

The pages I used for testing where:
https://www.heise.de/sso/login/
https://qis.fh-aachen.de (particularly bad due to the way Firefox focusses on the input field)

The videos have about 20 fps due to my recording software, but it's enough to see the flickering.

PS:
I'm very sorry for creating this comment twice before it was finished (I pressed Enter instead of STRG+Enter)

**Comment by paulverbeke** [comment]
I must say that on my end neither Bitwarden nor Dashlane works reliably using Autofill service on:
Firefox version 88.1.3
Firefox Beta version 89.0.0-beta.7
I must close and reopen the browser several times until the autofill popup is displayed (if it's even displayed at all).

However these password managers works perfectly using Autofill service on:
Chrome version 90.0.4430.210
Firefox Focus version 8.15.3
Firefox Nightly version 90.0a1

I must also concur with @Djfe on the fact that autofill popups are almost always rendered over the input forms, directly or after the fact because Firefox will scroll to the field and the popup doesn't follow the scroll. This behavior happens with both my password managers in all of the mentioned Firefox builds:
Firefox version 88.1.3
Firefox Beta version 89.0.0-beta.7
Firefox Nightly version 90.0a1
Firefox Focus version 8.15.3

But in Chrome version 90.0.4430.210, with these password managers, the autofill popups will correctly be rendered below the corresponding field.

(besides why is Dashlane not in the table ?)

Appart from this I must say that you've all done a great work with Fenix. I finally started to recommend Firefox (all the way!) to my relatives 😎💙 !
Thanks !!

**Comment by paulverbeke** [comment]
It's been several months and a lot of versions and the autofill is still not working reliably.

No matter the version this bug is present in:
Firefox stable
Firefox Beta

Not happening in:
Chrome
Firefox Focus
Firefox Nightly

*all downloaded from PlayStore

There's already a lot of issues opened related to this:
#14633
#19981
#20065: this one is proposing a new context menu to workaround this problem

Based on all my tests and all the comments I found, and exactly like @Djfe said:
We're guessing "that this is caused during the build of beta and release.
So it cannot be fixed in Nightly since it's not happening there"

Can the team please acknowledge this problem ? It's been at least a year. I even contacted Firefox Support on Twitter who instructed me to open an issue on this repo ^^'

**Comment by Djfe** [comment]
I know this doesn't add anything new to the issue, but I'm severely frustrated by now.
We feel ignored here :(

Please take a look at this bug

Regards

**Comment by Udayanithig** [comment]
Yes, it's super annoying. Still, firefox has no fix for this issue. And it forced me to use the Vivaldi browser. 💔

**Comment by dabat** [comment]
For my part, I am quite puzzled about how these password manager issues have remained unfixed for so long. 😢 

this is such a core part of everyday use of a browser application, I cannot understand how this sits at severity 2.
I would have expected this to be top priority.

[Brave browser](https://play.google.com/store/apps/details?id=com.brave.browser) works flawlessly (being Chromium-based) and so I have switched to Brave for daily use.

P.S. I will say that I really appreciate the extension support that Firefox has... DarkReader is super helpful in low-light 👍🏻

**Comment by abodea** [comment]
Hello guys, I will start doing some tests regarding Lastpass, and based on my results, I will try to open a new ticket to LastPass to see if there is something we can do on our side to make it work.

**Comment by Djfe** [comment]
Hi :) 
Maybe they have an idea, but this affects all password managers on Android. but only with Firefox release and beta. every(!) other app works just fine. Only Firefox behaves weirdly.
for example: keepass is also affected (at least)

Bitwarden managed to produce a workaround only for Firefox afaik.

**Comment by cadeyrn** [comment]
> but this affects all password managers on Android

I cannot confirm that it affects *all* password managers on Android. I use Firefox Lockwise as password manager and don't have any problems with it, neither in Firefox Nightly nor in the stable release of Firefox.

**Comment by Udayanithig** [comment]
Nope, most commercial password managers are affected by autofill issues on firefox. Very few of them works on firefox. I use Enpass and it still has auto-fill issues.

**Comment by paulverbeke** [comment]
Dashlane and Bitwarden both have this issue. The Bitwarden  workaround is really not consistent as you have to switch apps several times before the input field can be detected

**Comment by abodea** [comment]
Hi, @paulverbeke and @Djfe!

Can you please let me know if your problem is regarding LastPass or any other password managers (and let me know about them)?

Thank you,
Andrei

**Comment by paulverbeke** [comment]
Hi @abodea  thanks for taking the time to investigate this.
So to make this exhaustive I tested everything again, each time enabling the corresponding Autofill Service in my OnePlus 8T Android 11:

- [Bitwarden](https://www.dropbox.com/s/0yf5b5zjoh2ukmn/21-08-18-18-23-37%20-%20Bitwarden.mp4?dl=0): nothing when focusing on the fields, I have to close/open Firefox several times to make the popup appear
- [Dashlane](https://www.dropbox.com/s/pexxuxs93z7p928/21-08-18-18-29-04%20-%20Dashlane.mp4?dl=0): basically the same thing, except that it needs far less close/open of Firefox to appear
- NordPass (no video here as nothing to see of interest): nothing, didn't work at all, popup never appear. I had to also enable Accessibility Service for NordPass and then it worked fine like expected (popup blinking a lot but I don't know who is at fault here)
- [1Password](https://www.dropbox.com/sh/a35j58n7zmtuq0g/AADejbxbq0Dm8t9HsRz5BU9Pa?dl=0): works but only once you focus a second input field. After each new page load you need to focus email, then password (or reversed) and then popup appears.

I sincerely hope it'll be enough details.
Let me know if you need anything else

**Comment by rugk** [comment]
@paulverbeke AFAIK you can also upload videos on GitHub. Just drag and drop them. Third-party links tend tom get lost and then return 404 at some point of time…

**Comment by Djfe** [comment]
Hi @abodea,

I'm using enpass right now. OnePlus 3T, Havoc OS (Android 10 Custom Rom)

[Beta](http://www.youtube.com/watch?v=6JW8j1NG9ms "Beta") vs [Nightly](http://www.youtube.com/watch?v=pLSv8YjuMj8 "Nightly") (Beta flickers and moves the viewport around, Nightly only moves the viewport around)
These videos were recorded a couple of months ago, I would need to download them from youtube to reupload them to github. (in-case you are wondering, rugk)

Moving the view port after the click leads to enpass drawing it's tooltip at the wrong coordinates, because Firefox gives enpass the old coordinates of the input field on the screen instead of the new ones after moving the view port.
This can even lead to enpass drawing it's tooltip over the keyboard, because Firefox tells enpass that the input field is at the position of the keyboard.

The flickering bug only happens when relying on the Android Auto Fill Framework.
If I enable enpass as an accessibility service instead, then there is no flickering. The viewport still moves around though so it's still drawn at the wrong coordinates most of the time.

I also tested lastpass now (for the first time):
The autofill variant doesn't show up at all for me (while Chrome works fine)
The accessibility framework is working (but again: the viewport in Firefox moves)

**Comment by Udayanithig** [comment]
To be honest even firefox's "autofill in other apps" not working on firefox release nor beta.

**Comment by abodea** [comment]
@paulverbeke I've been busy with other tasks.
I started testing and I will update here once I move forward.
Today I reported this https://github.com/mozilla-mobile/fenix/issues/20987 as I was not able to make the autofill pop-up appear.
Started with Bitwarden and I will test all the APPS mentioned in the comments above.

**Comment by Djfe** [comment]
@abodea awesome
Pls keep in mind, that most issues we have only surface in beta and release but never in Nightly. (and on updates: when version 93 gets to beta status, then the bug will surface in beta where it never was as 93 nightly)

**Comment by abodea** [comment]
> @abodea awesome
> Pls keep in mind, that most issues we have only surface in beta and release but never in Nightly. (and on updates: when version 93 gets to beta status, then the bug will surface in beta where it never was as 93 nightly)

Thank you1 I will keep this in mind and as time allows will check on the RC and Beta too.

**Comment by paulverbeke** [comment]
@abodea thank you for your work.
Regarding Dashlane, [here's another recording](https://www.dropbox.com/s/dnrgg8y1vy9c3qy/21-08-25-00-00-39%20-%20dashlane%20%282%29.mp4?dl=0), with inline Keyboard Autofill enabled, showing you that the popup is not consistently appearing at first touch on the field

**Comment by Udayanithig** [comment]
Hi @abodea, The autofill is still present on the Firefox 93 release. I have also attached a Dropbox link for reference.
[Video Link](https://www.dropbox.com/s/ddx6l8jyjkkmmdh/20210917_101027.mp4?dl=0)

**Comment by bonyjose** [comment]
Inline autofill is not consistently working with Bitwarden, problem seems to be appearing and stable and beta builds, nightly seems to be working fine.
https://github.com/mozilla-mobile/fenix/issues/19685

**Comment by JpegXguy** [comment]
No difference between nightly or normal versions for me. Bitwarden autofill using the Android Autofill framework is very inconsistent for me in Fenix

- Sometimes the pop up to select an account doesn't appear and i have to tap away from the form field and back again ~6 times before it will appear.
- If the pop up appears and I tap on it to unlock Bitwarden, after selecting an account, it still doesn't get filled in

Using the old accessibility service method for autofilling (with the notification shade icon) used to work but for other reasons Bitwarden is not letting me enable the accessibility service. Ugh.

Also, we shouldn't need to use the accessibility service. Firefox it the only app I have a problem autofilling in.

**Comment by JpegXguy** [comment]
I was able to get Bitwarden to turn on its accessibility service be reinstalling it. Technically it could stop working again though.
Let's see how it goes with the official Autofill API though. Interesting Bugzilla thread. I'll keep an eye on it, thanks

**Comment by paulverbeke** [comment]
Hi everyone, today on stable 93.2, I tested Bitwarden, Dashlane, 1Password and they all works perfectly and reliably, directly on the first focus, and without any workaround ❤️ thanks a lot for all this teamwork cross project !

I must still note 2 things:
- for all 3 of them, once you autofilled a credential then emptied the fields (because it wasn't the credential you wanted), you cannot get any password manager to work  until you refreshed the page
- Nordpass is still not poping at all with autofill services, but I guess it must be on their side 😋

**Comment by Dimas-sc** [comment]
No love to Keepass2Android? 😥

**Comment by soredake** [comment]
Need to wait for new version https://github.com/PhilippC/keepass2android/commit/0564a103e15ad0d3f4fde604bef2b9d00b106a00

**Comment by Udayanithig** [comment]
Still bitwarden is not working for me, but the inline autofill is consistent but on the other hand, credentials are not filling on the login page.

https://user-images.githubusercontent.com/76005728/140646801-c7c5d39c-6ecd-4d60-8754-8f1b32533348.mp4

**Comment by paulverbeke** [comment]
According to your capture I'm guessing several things are happening here:
- 1st, based on the fact that you had to unlock your bitwarden vault   on both pages, I suppose you configured bitwarden "Vault Timeout" to "immediately" and "Vault Timeout Action" to "Lock". Personally I have it set on 15min.
- 2nd, the inline autofill entry "My Vault" that opens bitwarden Fullscreen to allow choosing credentials to autofill (or allow unlocking vault then choosing credential) never worked AFAIK on Firefox Android. The workaround is to have your vault setup to stay unlocked a bit like at least 1 minute, tap "My Vault", unlock the vault, and then go back instead of choosing your credential on this page, then you'll find your credential entry in the inline autofill, tap it and it works![](https://i.imgur.com/eokEjpy.jpg)

**Comment by makotokato** [comment]
@Udayanithig 
I guess that this is same issue as https://bugzilla.mozilla.org/show_bug.cgi?id=1693152. When switching activity to Bitwarden, it doesn't fill username and password well. I will look it.

**Comment by Udayanithig** [comment]
Thanks @makotokato

**Comment by ChrisG683** [comment]
Thank you from me as well!  This bug has been plaguing Firefox for Android for what feels like 2 years now, I will celebrate the day we can finally be past this bug haha.

**Comment by Jwpe** [comment]
I've been having issues with LastPass and Firefox for Android since the release of this new implementation. I've got into the habit of just copying passwords directly from the LastPass app instead of trying to rely on the autofill behaviour.  However, I've just tested it again after finding this GitHub issue, and the behaviour is very odd. 

It doesn't completely fail, but the LastPass overlay doesn't appear when focussing a password field for the first time. After de-selecting and then focussing the field again, the overlay flickers briefly into sight and then disappears again before any credentials can be selected. Repeating the process of de-selecting and and focussing the password field several times, the same thing happens - the overlay briefly appears and then disappears again. Eventually, I can get the overlay to appear permanently after several repetitions. 

If I can make a GIF of this to demonstrate, I'll try to do so. I don't have this issue wth autofill in other Android apps.

**Comment by tommcc** [comment]
Thanks everyone. I have been following this issue for a while, and have been both a) pleased that implementation has improved but b) can confirm I've been having the _exact_ same "flicker" behavior that @Jwpe perfectly describes (Android 12, LastPass).

**Comment by rogervn** [comment]
I can also second the exact same behaviour as @Jwpe. I actually gave up on Firefox a while back because of this bug. The reason I'm following this thread is to find out if I can come back sometime in the future.

**Comment by abodea** [comment]
> I've been having issues with LastPass and Firefox for Android since the release of this new implementation. I've got into the habit of just copying passwords directly from the LastPass app instead of trying to rely on the autofill behaviour. However, I've just tested it again after finding this GitHub issue, and the behaviour is very odd.
> 
> It doesn't completely fail, but the LastPass overlay doesn't appear when focussing a password field for the first time. After de-selecting and then focussing the field again, the overlay flickers briefly into sight and then disappears again before any credentials can be selected. Repeating the process of de-selecting and focussing the password field several times, the same thing happens - the overlay briefly appears and then disappears again. Eventually, I can get the overlay to appear permanently after several repetitions.
> 
> If I can make a GIF of this to demonstrate, I'll try to do so. I don't have this issue with autofill in other Android apps.

Thank you for this message!
Sorry for the delayed response but I have been changing my project and I had a lot of work on the iOS side.
I just re-tested with my phone for few websites and it worked very well this time on the latest Nightly build with LastPass.

Do you still have that weird behavior or is it working for you now?
Asking because I do some more testing and I will try to talk with the Lastpass CC as they will help us to debug this.

Thanks,
Andrei

**Comment by makotokato** [comment]
@abodea 
As long as I look this screencast, it seems to be same issue as Bitwarden etc that is autofill compat mode... If you use beta, there is workaround (https://github.com/mozilla-mobile/fenix/issues/9773#issuecomment-945004703).

**Comment by Udayanithig** [comment]
> Still bitwarden is not working for me, but the inline autofill is consistent but on the other hand, credentials are not filling on the login page.
> Screenrecorder-2021-11-07-18-49-25-340.mp4

Finally fixed on Firefox beta 102. 🎉

**Comment by cpeterso** [comment]
> Finally fixed on Firefox beta 102. 🎉

Thanks for testing. In that case, I'll close this bug as fixed. If there are other password manager bugs, please file bug reports in Bugzilla's "Fenix > Privacy" component here:

https://bugzilla.mozilla.org/describecomponents.cgi?product=Fenix
